pkg/solana/chain.go

```diff
@@ -90,6 +92,7 @@ type chain struct {
 	services.StateMachine
```

Comment:
Using the following combined as service state helps clean up boilerplate:

```
	services.Service
	engine *services.Engine
```

And in the init:

```
	p.Service, p.engine = services.Config{
		Name: "LogPollerService",
		NewSubServices: func(lggr logger.Logger) []services.Service {
			p.loader = NewEncodedLogCollector(client, p, lggr)

			return []services.Service{p.loader}
		},
		Start: p.start,
		Close: p.close,
	}.NewServiceEngine(lggr)
	p.lggr = p.engine.SugaredLogger
```

Reply:
Is this comment in reference to `Chain` in this file, or to `LogPoller` in the logpoller file? The example you give looks like LogPoller but you left it on the `Chain` interface.

Reply:
Ahh my bad. It references `LogPoller` and may not be relevant. It looks like you did it this way so ignore my comment.